cmd/awsDataGather/main.go

```diff
@@ -0,0 +1,138 @@
+package main
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"time"
+
+	"github.com/red-hat-storage/ocs-osd-deployer/utils"
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	utilruntime "k8s.io/apimachinery/pkg/util/runtime"
+	clientgoscheme "k8s.io/client-go/kubernetes/scheme"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/client/config"
+	"sigs.k8s.io/controller-runtime/pkg/log/zap"
+)
+
+func main() {
+	ctrl.SetLogger(zap.New(zap.UseDevMode(true)))
+	log := ctrl.Log.WithName("main")
+
+	mainContext := context.Background()
+
+	namespace, found := os.LookupEnv("NAMESPACE")
+	if !found {
+		fmt.Fprintf(os.Stderr,
+			"NAMESPACE environment variable not found\n")
+		os.Exit(1)
+	}
+	podName, found := os.LookupEnv("POD_NAME")
+	if !found {
+		fmt.Fprintf(os.Stderr,
+			"POD_NAME environment variable not found\n")
+		os.Exit(1)
+	}
+
+	// We need the deployment name to use as an owner reference.
+	// However, the deployment name cannot be passed as an environment variable,
+	// so we derive it from the pod name.
+	deploymentName, err := utils.DeploymentNameFromPodName(podName)
+	if err != nil {
+		fmt.Fprintf(
+			os.Stderr,
+			"Could not determine deployment name from pod name (%s)",
+			podName)
+		os.Exit(1)
+	}
+
+	log.Info("Setting up k8s client")
+	var options client.Options
+	options.Scheme = runtime.NewScheme()
+	utilruntime.Must(clientgoscheme.AddToScheme(options.Scheme))
+
+	k8sClient, err := client.New(config.GetConfigOrDie(), options)
+	if err != nil {
+		log.Error(err, "error creating client")
+		os.Exit(1)
+	}
+
+	// This will later be used as the OwnerReference for the data ConfigMap
+	var deployment appsv1.Deployment
+	err = k8sClient.Get(mainContext, client.ObjectKey{
+		Name:      deploymentName,
+		Namespace: namespace,
+	}, &deployment)
+	if err != nil {
+		log.Error(err, fmt.Sprintf("Failed to find deployment '%s' in namespace '%s'", deploymentName, namespace))
+		os.Exit(1)
+	}
+
+	var backoff time.Duration = 1
+	var maxSleep time.Duration = 5
+	for {
+		var sleep time.Duration
+		log.Info("Gathering AWS data")
+		if err := gatherAndSaveData(utils.IMDSv1Server, deployment, k8sClient, mainContext); err == nil {
+			log.Info("AWS data gathering successfully completed!")
+			sleep = maxSleep
+			// Reset the backoff counter since data gathering succeeded.
+			backoff = 1
+		} else {
+			log.Error(err, "Failed to gather AWS data")
+			sleep = backoff
+			backoff = 2 * backoff
+			if backoff > maxSleep {
+				backoff = maxSleep
+			}
+		}
+		log.Info("Sleeping for %d minutes before next the fetch...", sleep)
+		time.Sleep(sleep * time.Minute)
```

Comment:
This is supposed to be in Seconds not minutes. 
I guess this change got in by accident via a suggestion I made on a previous review.

```suggestion
		time.Sleep(sleep * time.Second)
```

Reply:
I did this intentionally. Why does it need to be in seconds?
As it is written, backoff will go 1, 2, 4, 5 minutes.

Reply:
I don't want to get stuck on this point, I changed it to seconds.

Reply:
Even a single minute is a very long time to wait when you have an error and wants to repeat.
Thank you for changing to seconds